scripts/linters/js_ts_linter.py

```diff
@@ -1139,6 +1139,109 @@ def _check_comments(self):
             summary_messages.append(summary_message)
         return summary_messages
 
+    def _check_angular_services_index(self):
+        """Finds all @Injectable classes and makes sure that they are added to
+            Oppia root and Angular Services Index.
+
+        Returns:
+            all_messages: str. All the messages returned by the lint checks.
+        """
+        def get_injectable_class_name(file_content):
+            """Extarcts the class name from a file that has an Injectable
+                class.
+
+            Args:
+                file_content: str. File content of the file that has an
+                    Injectable class.
+
+            Returns:
+                tuple(str, str). A two-tuple of class name and class name in
+                camelCase.
+            """
+
+            # Going through all ts files and looking for @Injectable classes.
+            class_name = file_content.split(
+                '@Injectable({')[1].split(
+                    'export class ')[1].split('{')[0].replace(' ', '')
+            if 'extends' in class_name:
+                class_name = class_name.split('extends')[0]
+            if 'implements' in class_name:
+                class_name = class_name.split('implements')[0]
+            if '<' in class_name:
+                class_name = class_name.split('<')[0]
+            return class_name
+
+        angular_services_index_path = (
+            './core/templates/services/angular-services.index.ts')
+        injectables_to_ignore = [
+            'ExplorationStats',
+            'InteractionAttributesExtractorService',
+            'NormalizeWhitespacePunctuationAndCasePipe\n',
+            'MockCsrfTokenService',
+            'StatePropertyService',
+            'StateNextContentIdIndexService\n',
+            'SubtitledUnicodeObjectFactory',
+            'UpgradedServices'
+        ]
+        if self.verbose_mode_enabled:
+            python_utils.PRINT('Starting Angular Services Index file check')
+            python_utils.PRINT('----------------------------------------')
+        angular_services_index = self.file_cache.read(
+            angular_services_index_path)
+        summary_messages = []
+        total_error_count = 0
+        total_files_checked = 0
+        for file_path in self.ts_files:
+            file_content = self.file_cache.read(file_path)
+            if '@Injectable({' in file_content:
+                total_files_checked += 1
+                class_name = get_injectable_class_name(file_content)
+                if class_name in injectables_to_ignore:
+                    continue
+                import_statement = 'import { %s' % class_name
```

Comment:
optional: Should we use `import {  %s  }`, just to ensure that we are tracking the full class name? [I'm fine with this one as well]

Reply:
I have left it as it is for now.